### PR TITLE
ACS-1778 Common transform routing code for Repo and t-router

### PR DESCRIPTION
**Pull request by alandavis**
Main change is the removal of code from CombinedConfig into alfresco-transform-model as CombinedTransformConfig.The Code within LocalCombinedConfig has now moved to CombinedConfig as there is only one type of config now.

**Depends on a new version of alfresco-transform-model (see https://github.com/Alfresco/alfresco-transform-model/pull/93)**

### CODE REVIEW

File: repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java [hunk 1]
```diff
@@ -113,9 +74,7 @@ public CombinedConfig(Log log)
             protected void readJson(JsonNode jsonNode, String readFrom, String baseUrl)
             {
                 TransformConfig transformConfig = jsonObjectMapper.convertValue(jsonNode, TransformConfig.class);
-                transformConfig.getTransformOptions().forEach((key, map) -> combinedTransformOptions.put(key, map));
-                transformConfig.getTransformers().forEach(transformer -> combinedTransformers.add(
-                        new TransformAndItsOrigin(transformer, baseUrl, readFrom)));
+                addTransformConfig(transformConfig, readFrom, baseUrl);
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java:77]: addTransformConfig(TransformConfig) is a new method to make it easier for the t-router to use the common CombinedTransformConfig class in the model.

File: repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java [hunk 2]
```diff
@@ -255,6 +214,12 @@ private String getErrorMessage(HttpEntity resEntity) throws IOException
         return message;
     }
 
+    @Override
+    protected boolean isPassThroughTransformName(String name)
+    {
+        return name.equals(LocalPassThroughTransform.NAME);
+    }
+
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java:222]: Method came from the sub class LocalCombinedConfig which has now been removed.

File: repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java [hunk 3]
```diff
@@ -273,277 +238,7 @@ public void register(TransformServiceRegistryImpl registry)
         data.setTEngineCount(tEngineCount);
         data.setFileCount(configFileFinder.getFileCount());
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java:240]: The following code is now in the common CombinedTransformConfig class of the transform model for use by both the Repo and t-router.

File: repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java [hunk 4]
```diff
@@ -273,277 +238,7 @@ public void register(TransformServiceRegistryImpl registry)
...
-        combinedTransformers = removeInvalidTransformers(combinedTransformers, registry);
-        combinedTransformers = sortTransformers(combinedTransformers, registry);
-        addWildcardSupportedSourceAndTarget(combinedTransformers);
-
-        combinedTransformers.forEach(transformer ->
-            registry.register(transformer.transformer, combinedTransformOptions,
-                    transformer.baseUrl, transformer.readFrom));
-    }
-
-    /**
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java:285]: The following code is now in the common CombinedTransformConfig class of the transform model for use by both the Repo and t-router.

File: repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java [hunk 5]
```diff
@@ -47,58 +44,22 @@
...
+ * The caller should make calls to {@link #addRemoteConfig(List, String)}, {@link #addLocalConfig(String)} or
+ * {@link #addTransformConfig(TransformConfig, String, String)} followed by a call to
+ * {@link #register(TransformServiceRegistryImpl)}.
  *
  * @author adavis
  */
-public class CombinedConfig
+public class CombinedConfig extends CombinedTransformConfig
 {
     private final Log log;
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/transform/client/registry/CombinedConfig.java:62]: The following code is now in the common CombinedTransformConfig class of the transform model for use by both the Repo and t-router.

**Review by PunchedChimera — APPROVED** [review]
Looks good to me, all functionality/code removed has been added to [alfresco-transform-model/pull/93](https://github.com/Alfresco/alfresco-transform-model/pull/93)

**Pull request merged**